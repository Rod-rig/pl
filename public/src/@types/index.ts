export * from './ITable';
export * from './IMatch';
export * from './ITournament';
