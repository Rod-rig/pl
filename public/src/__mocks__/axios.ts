import config from '../config/config';
import {matchListMock, scheduleMock, tableMock, tournamentListMock} from './';

export default {
  get: jest.fn((url) => {
    const rootUrl = `${config.apiUrl}/en`;
    const urlWithoutKey = url.split('?')[0];
    if (urlWithoutKey === `${rootUrl}/tournaments.json`) {
      return Promise.resolve({data: tournamentListMock});
    } else if (urlWithoutKey === `${rootUrl}/tournaments/test/standings.json`) {
      return Promise.resolve({data: tableMock});
    } else if (urlWithoutKey === `${rootUrl}/tournaments/test_id/results.json`) {
      return Promise.resolve({data: matchListMock});
    } else if (urlWithoutKey === `${rootUrl}/schedules/2018-07-07/schedule.json`) {
      return Promise.resolve({data: scheduleMock});
    } else {
      return Promise.reject('No such url');
    }
  }),
};
