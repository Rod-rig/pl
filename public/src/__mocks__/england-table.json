[
  {
    "position": 1,
    "w": 27,
    "d": 3,
    "l": 1,
    "goals for": 88,
    "goals against": 21,
    "points": 84,
    "teamName": "Manchester City"
  },
  {
    "position": 2,
    "w": 21,
    "d": 5,
    "l": 5,
    "goals for": 60,
    "goals against": 23,
    "points": 68,
    "teamName": "Manchester United"
  },
  {
    "position": 3,
    "w": 19,
    "d": 9,
    "l": 4,
    "goals for": 75,
    "goals against": 35,
    "points": 66,
    "teamName": "Liverpool"
  }
]