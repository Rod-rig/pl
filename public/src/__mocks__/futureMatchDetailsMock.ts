/* tslint:disable: object-literal-key-quotes object-literal-sort-keys*/
export const futureMatchDetailsMock = {
  generated_at: "2019-08-04T16:21:35+00:00",
  schema:
    "http://schemas.sportradar.com/bsa/soccer/v1/json/endpoints/soccer/match_summary.json",
  sport_event: {
    id: "sr:match:18427304",
    scheduled: "2019-08-09T19:00:00+00:00",
    start_time_tbd: false,
    tournament_round: { type: "group", number: 1, phase: "regular season" },
    season: {
      id: "sr:season:66441",
      name: "Premier League 19/20",
      start_date: "2019-08-09",
      end_date: "2020-05-18",
      year: "19/20",
      tournament_id: "sr:tournament:17",
    },
    tournament: {
      id: "sr:tournament:17",
      name: "Premier League",
      sport: { id: "sr:sport:1", name: "Soccer" },
      category: { id: "sr:category:1", name: "England", country_code: "ENG" },
    },
    competitors: [
      {
        id: "sr:competitor:44",
        name: "Liverpool FC",
        country: "England",
        country_code: "ENG",
        abbreviation: "LFC",
        qualifier: "home",
      },
      {
        id: "sr:competitor:263",
        name: "Norwich City",
        country: "England",
        country_code: "ENG",
        abbreviation: "NOR",
        qualifier: "away",
      },
    ],
    venue: {
      id: "sr:venue:579",
      name: "Anfield",
      capacity: 54074,
      city_name: "Liverpool",
      country_name: "England",
      map_coordinates: "53.430622,-2.960919",
      country_code: "ENG",
    },
  },
  sport_event_conditions: {
    venue: {
      id: "sr:venue:579",
      name: "Anfield",
      capacity: 54074,
      city_name: "Liverpool",
      country_name: "England",
      map_coordinates: "53.430622,-2.960919",
      country_code: "ENG",
    },
  },
  sport_event_status: { status: "not_started" },
};
