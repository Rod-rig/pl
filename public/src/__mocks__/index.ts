export * from "./futureMatchDetailsMock";
export * from "./matchDetailsMock";
export * from "./matchListMock";
export * from "./scheduleByDateMock";
export * from "./scheduleByTournamentId";
export * from "./tableMock";
export * from "./tournamentListMock";
export * from "./userMock";
export * from "./predictions";
