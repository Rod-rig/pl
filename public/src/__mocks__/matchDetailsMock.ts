/* tslint:disable: object-literal-key-quotes object-literal-sort-keys*/
export const matchDetailsMock = {
  generated_at: "2019-07-27T09:57:47+00:00",
  schema:
    "http://schemas.sportradar.com/bsa/soccer/v1/json/endpoints/soccer/match_summary.json",
  sport_event: {
    id: "sr:match:14735957",
    scheduled: "2018-08-10T19:00:00+00:00",
    start_time_tbd: false,
    tournament_round: {
      type: "group",
      number: 1,
      phase: "regular season",
    },
    season: {
      id: "sr:season:54571",
      name: "Premier League 18/19",
      start_date: "2018-08-10",
      end_date: "2019-05-13",
      year: "18/19",
      tournament_id: "sr:tournament:17",
    },
    tournament: {
      id: "sr:tournament:17",
      name: "Premier League",
      sport: {
        id: "sr:sport:1",
        name: "Soccer",
      },
      category: {
        id: "sr:category:1",
        name: "England",
        country_code: "ENG",
      },
    },
    competitors: [
      {
        id: "sr:competitor:35",
        name: "Manchester United",
        country: "England",
        country_code: "ENG",
        abbreviation: "MNU",
        qualifier: "home",
      },
      {
        id: "sr:competitor:31",
        name: "Leicester City",
        country: "England",
        country_code: "ENG",
        abbreviation: "LEI",
        qualifier: "away",
      },
    ],
    venue: {
      id: "sr:venue:9",
      name: "Old Trafford",
      capacity: 75635,
      city_name: "Manchester",
      country_name: "England",
      map_coordinates: "53.463150,-2.291444",
      country_code: "ENG",
    },
  },
  sport_event_conditions: {
    referee: {
      id: "sr:referee:52261",
      name: "Marriner, Andre",
      nationality: "England",
      country_code: "ENG",
    },
    referee_assistants: [
      {
        type: "first_assistant_referee",
        id: "sr:referee:79890",
        name: "Ledger, Scott",
        nationality: "England",
        country_code: "ENG",
      },
      {
        type: "second_assistant_referee",
        id: "sr:referee:88710",
        name: "Long, Simon",
        nationality: "England",
        country_code: "ENG",
      },
    ],
    venue: {
      id: "sr:venue:9",
      name: "Old Trafford",
      capacity: 75635,
      city_name: "Manchester",
      country_name: "England",
      map_coordinates: "53.463150,-2.291444",
      country_code: "ENG",
    },
    attendance: 74439,
    weather_info: {
      pitch: "good",
      weather_conditions: "good",
    },
  },
  sport_event_status: {
    status: "closed",
    match_status: "ended",
    home_score: 2,
    away_score: 1,
    winner_id: "sr:competitor:35",
    period_scores: [
      {
        home_score: 1,
        away_score: 0,
        type: "regular_period",
        number: 1,
      },
      {
        home_score: 1,
        away_score: 1,
        type: "regular_period",
        number: 2,
      },
    ],
  },
  statistics: {
    teams: [
      {
        id: "sr:competitor:35",
        name: "Manchester United",
        abbreviation: "MNU",
        qualifier: "home",
        statistics: {
          ball_possession: 46,
          shots_blocked: 1,
          shots_on_target: 6,
          throw_ins: 19,
          fouls: 11,
          goal_kicks: 7,
          shots_saved: 3,
          free_kicks: 9,
          offsides: 4,
          corner_kicks: 2,
          shots_off_target: 1,
          injuries: 1,
          yellow_cards: 2,
        },
        players: [
          {
            id: "sr:player:69378",
            name: "De Gea, David",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:19352",
            name: "Darmian, Matteo",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:606346",
            name: "Bailly, Eric",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:143334",
            name: "Lindelof, Victor",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:190839",
            name: "Shaw, Luke",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 1,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:243623",
            name: "Fred",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 1,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:285949",
            name: "Pereira, Andreas",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 1,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:111802",
            name: "Pogba, Paul",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 1,
            penalty_goals_scored: 1,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:34120",
            name: "Sanchez, Alexis",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 1,
          },
          {
            id: "sr:player:903830",
            name: "Rashford, Marcus",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:28367",
            name: "Mata, Juan",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 1,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:78893",
            name: "Lukaku, Romelu",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:908372",
            name: "McTominay, Scott",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:21764",
            name: "Fellaini, Marouane",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
        ],
      },
      {
        id: "sr:competitor:31",
        name: "Leicester City",
        abbreviation: "LEI",
        qualifier: "away",
        statistics: {
          ball_possession: 54,
          free_kicks: 15,
          throw_ins: 20,
          shots_on_target: 4,
          offsides: 2,
          fouls: 7,
          goal_kicks: 4,
          shots_blocked: 6,
          corner_kicks: 5,
          shots_saved: 4,
          shots_off_target: 4,
          yellow_cards: 1,
        },
        players: [
          {
            id: "sr:player:1076",
            name: "Schmeichel, Kasper",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:308526",
            name: "Amartey, Daniel",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 1,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:20375",
            name: "Morgan, Wes",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:149380",
            name: "Maguire, Harry",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:754594",
            name: "Chilwell, Benjamin",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:221200",
            name: "Pereira, Ricardo",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:795266",
            name: "Ndidi, Onyinye Wilfred",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:30127",
            name: "Silva, Adrien",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:356398",
            name: "Maddison, James",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:355152",
            name: "Gray, Demarai",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:359642",
            name: "Iheanacho, Kelechi",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:123228",
            name: "Ghezzal, Rachid",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:173827",
            name: "Vardy, Jamie",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 1,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:32690",
            name: "Iborra, Vicente",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
        ],
      },
    ],
  },
};
