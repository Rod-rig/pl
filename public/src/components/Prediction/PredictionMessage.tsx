import {
  Button,
  createStyles,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  Theme,
  WithStyles,
  withStyles,
} from "@material-ui/core";
import { green } from "@material-ui/core/colors";
import Done from "@material-ui/icons/Done";
import * as React from "react";
import { dict } from "../../dict";

const styles = ({ palette }: Theme) =>
  createStyles({
    btn: {
      color: palette.common.white,
    },
    btnWrap: {
      justifyContent: "center",
    },
    dialog: {
      backgroundColor: green[600],
    },
    icon: {
      color: palette.common.white,
      display: "block",
      fontSize: 100,
      margin: "auto",
    },
    text: {
      color: palette.common.white,
      fontSize: "2.25rem",
      textTransform: "uppercase",
    },
  });

interface IProps extends WithStyles<typeof styles> {
  handleClose: () => void;
  open: boolean;
}

export const PredictionMessage = withStyles(styles)((props: IProps) => {
  const { classes, handleClose, open } = props;
  return (
    <Dialog
      classes={{
        paper: classes.dialog,
      }}
      open={open}
      keepMounted={true}
      onClose={handleClose}
    >
      <DialogContent>
        <Done className={classes.icon} />
        <DialogContentText className={classes.text}>
          {dict.success}
        </DialogContentText>
      </DialogContent>
      <DialogActions className={classes.btnWrap}>
        <Button className={classes.btn} onClick={handleClose}>
          {dict.continueText}
        </Button>
      </DialogActions>
    </Dialog>
  );
});
