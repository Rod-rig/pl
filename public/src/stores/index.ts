export * from './TableStore/TableStore';
export * from './MatchListStore/MatchListStore';
