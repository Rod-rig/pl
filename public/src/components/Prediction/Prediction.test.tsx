import { CircularProgress, Typography } from "@material-ui/core";
import { mount } from "enzyme";
import * as React from "react";
import { Loader, Prediction } from "../";
import { ISportEvent } from "../../@types";
import { scheduleMock } from "../../__mocks__";
import { userStore } from "../../stores";
import { PredictionMessage } from "./PredictionMessage";

describe("Prediction", () => {
  const createPrediction = (
    isLoaded: boolean,
    data: ISportEvent[],
    isSuccessSubmit: boolean,
    isFetched: boolean,
    buttonWasClicked: boolean,
  ) => {
    const closeSuccessMsgMock = jest.fn();
    const fetchMatchesMock = jest.fn();
    const handleChangeMock = jest.fn();
    const handleDateChangeMock = jest.fn();
    const handleSubmitMock = jest.fn();
    const handleTournamentChangeMock = jest.fn();
    const setCurrentDateMock = jest.fn();
    const setTournamentIdMock = jest.fn();
    const setMatchesMock = jest.fn();
    return (
      <Prediction
        store={{
          buttonWasClicked,
          cache: {},
          closeSuccessMsg: closeSuccessMsgMock,
          currentDate: "",
          dates: [],
          fetchMatches: fetchMatchesMock,
          handleChange: handleChangeMock,
          handleDateChange: handleDateChangeMock,
          handleSubmit: handleSubmitMock,
          handleTournamentChange: handleTournamentChangeMock,
          isFetched,
          isLoaded,
          isSuccessSubmit,
          matches: data,
          setCurrentDate: setCurrentDateMock,
          setMatches: setMatchesMock,
          setTournamentId: setTournamentIdMock,
          tournamentId: "",
          tournaments: {},
        }}
      />
    );
  };
  const comp = createPrediction(true, scheduleMock, false, true, false);

  it("should render loader", () => {
    const notRenderedComp = mount(
      createPrediction(false, [], false, true, false),
    );
    expect(notRenderedComp.find(Loader)).toHaveLength(1);
  });

  it("should render submit button with loader", () => {
    const wrapper = mount(
      createPrediction(true, scheduleMock, false, true, true),
    );
    expect(wrapper.find(CircularProgress)).toHaveLength(1);
  });

  it("should render success message after form submit", () => {
    const wrapper = mount(
      createPrediction(true, scheduleMock, true, true, false),
    );
    expect(wrapper.find(PredictionMessage)).toHaveLength(1);
  });

  it("should show message when there are no matches", () => {
    const wrapper = mount(createPrediction(true, [], true, true, false));
    expect(wrapper.find(Typography)).toHaveLength(1);
  });

  it("should show loader when matches are fetching", () => {
    const wrapper = mount(createPrediction(true, [], true, false, false));
    expect(wrapper.find(Loader)).toHaveLength(1);
  });

  it("should show message for not logged in user", () => {
    userStore.isLoggedIn = undefined;
    const wrapper = mount(comp);
    expect(wrapper.find(Loader)).toHaveLength(1);
  });
});
