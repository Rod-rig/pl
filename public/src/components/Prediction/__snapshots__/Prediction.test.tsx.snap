// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PredictionForm should render correctly 1`] = `
<div>
  User is not authorized
</div>
`;
